newIDE/app/src/Profile/Subscription/SubscriptionChecker.js

```diff
@@ -0,0 +1,151 @@
+// @flow
+
+import * as React from 'react';
+import { Trans } from '@lingui/macro';
+import FlatButton from '../../UI/FlatButton';
+import Dialog, { DialogPrimaryButton } from '../../UI/Dialog';
+import Star from '@material-ui/icons/Star';
+import Favorite from '@material-ui/icons/Favorite';
+import AuthenticatedUserContext from '../AuthenticatedUserContext';
+import { Column, LargeSpacer, Line } from '../../UI/Grid';
+import {
+  sendSubscriptionCheckDialogShown,
+  sendSubscriptionCheckDismiss,
+} from '../../Utils/Analytics/EventSender';
+import { SubscriptionSuggestionContext } from './SubscriptionSuggestionContext';
+import Text from '../../UI/Text';
+
+export type SubscriptionCheckerInterface = {|
+  checkHasSubscription: () => boolean,
+|};
+
+type Props = {|
+  title: React.Node,
+  id:
+    | 'Disable GDevelop splash at startup'
+    | 'Debugger'
+    | 'Hot reloading'
+    | 'Preview over wifi',
+  onChangeSubscription?: () => Promise<void> | void,
+  mode: 'try' | 'mandatory',
+|};
+
+const styles = {
+  icon: { width: 40, height: 40, marginRight: 20 },
+  iconText: { flex: 1 },
+};
+
+const SubscriptionChecker = React.forwardRef<
+  Props,
+  SubscriptionCheckerInterface
+>(({ mode, id, title, onChangeSubscription }, ref) => {
+  const authenticatedUser = React.useContext(AuthenticatedUserContext);
+  const [dialogOpen, setDialogOpen] = React.useState<boolean>(false);
+  const { openSubscriptionDialog } = React.useContext(
+    SubscriptionSuggestionContext
+  );
+
+  const closeDialog = () => {
+    sendSubscriptionCheckDismiss();
+    setDialogOpen(false);
+  };
+
+  const checkHasSubscription = () => {
```

Comment:
the naming is a bit confusing.
it does not only check if there is a subscription it opens the dialog too.
Maybe `checkAndOpenSubscriptionDialog`?

Reply:
A truly accurate naming would read `checkSubscriptionAndOpenSubscriptionDialogIfNotEnough` but it's a bit cumbersome.
`openSubscriptionDialogIfUserIsUnsubscribed`?

Reply:
But I feel like having `checkUserHasSubscription` is simpler and not that misleading

Reply:
alright!